test/integration_test/common_test.go

```diff
@@ -70,11 +72,45 @@ func validateStorageCluster(
 	cluster *corev1alpha1.StorageCluster,
 	timeout, interval time.Duration,
 ) error {
-	// Validate expected portworx pods are running and no more, no less
-	return validateStorageClusterPods(cluster, timeout, interval)
+	var err error
+	cluster, err = k8s.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+
+	if err = validateStorageClusterPods(cluster, timeout, interval); err != nil {
+		return err
+	}
+
+	svc, err := k8s.Instance().GetService("portworx-service", cluster.Namespace)
+	pxEndpoint := ""
+	if err == nil {
+		pxEndpoint = svc.Spec.ClusterIP
+	}
+
+	conn, err := grpc.Dial(pxEndpoint, grpc.WithInsecure())
+	if err != nil {
+		return err
+	}
+	nodeClient := api.NewOpenStorageNodeClient(conn)
+	nodeEnumerateResp, err := nodeClient.Enumerate(context.Background(), &api.SdkNodeEnumerateRequest{})
+	if err != nil {
+		return err
+	}
+	expectedNodes := len(cluster.Spec.Nodes)
+	actualNodes := len(nodeEnumerateResp.GetNodeIds())
+	if actualNodes != expectedNodes {
+		return fmt.Errorf("expected nodes: %v. actual nodes: %v", expectedNodes, actualNodes)
+	}
 
-	// TODO: Validate expected number of portworx nodes are online (using sdk)
 	// TODO: Validate portworx is started with correct params. Check individual options
+	for _, n := range nodeEnumerateResp.GetNodeIds() {
+		nodeResp, err := nodeClient.Inspect(d.getContext(), &api.SdkNodeInspectRequest{NodeId: n})
```

Comment:
This is good. To start with, we should validate the storage cluster params against the response received from inspect. At the very least check the status is Online/Running from the response.